lib/pbench/test/unit/server/query_apis/test_datasets_contents.py

```diff
@@ -0,0 +1,412 @@
+from http import HTTPStatus
+
+import pytest
+
+from pbench.server.api.resources.query_apis.datasets.datasets_contents import (
+    DatasetsContents,
+)
+from pbench.server.database.models.datasets import Dataset
+from pbench.test.unit.server.query_apis.commons import Commons
+
+
+class TestDatasetsContents(Commons):
+    """
+    Unit testing for DatasetsContents class.
+    In a web service context, we access class functions mostly via the
+    Flask test client rather than trying to directly invoke the class
+    constructor and `post` service.
+    """
+
+    @pytest.fixture(autouse=True)
+    def _setup(self, client):
+        super()._setup(
+            cls_obj=DatasetsContents(client.config, client.logger),
+            pbench_endpoint="/datasets/contents",
+            elastic_endpoint="/_search",
+            payload={"run_id": "random_md5_string1", "parent": "/1-default"},
+            index_from_metadata="run-toc",
+        )
+
+    def test_with_no_index_document(self, client, server_config):
+        """
+        Check the DatasetsContents API when no index name is provided
+        """
+        # remove the last component of the pbench_endpoint
+        incorrect_endpoint = "/".join(self.pbench_endpoint.split("/")[:-1])
+        response = client.get(f"{server_config.rest_uri}{incorrect_endpoint}")
+        assert response.status_code == HTTPStatus.NOT_FOUND
+
+    def test_with_incorrect_index_document(self, client, server_config, pbench_token):
+        """
+        Check the Contents API when an incorrect index name is provided.
+        """
+        incorrect_endpoint = "/".join(self.pbench_endpoint.split("/")[:-1]) + "/test"
+        response = client.post(
+            f"{server_config.rest_uri}{incorrect_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.payload,
+        )
+        assert response.status_code == HTTPStatus.NOT_FOUND
+
+    def test_query(
+        self,
+        server_config,
+        query_api,
+        pbench_token,
+        build_auth_header,
+        find_template,
+        provide_metadata,
+    ):
+        """
+        Check the construction of Elasticsearch query URI and filtering of the
+        response body. Note that the mock set up by the attach_dataset fixture
+        matches the dataset name to the dataset's owner.
```

Comment:
This advice may still be relevant, although it's now buried several layers deep; but I think that in a pytest case we should focus on describing what the test case *does*, with indirect help like this in a second or third paragraph.

In particular, from reading the docstrings it should be clear how `test_query` and `test_subdirectory_query` differ.